src/shell/autoexec.cpp

```diff
@@ -507,9 +540,152 @@ void AutoExecModule::AddLine(const Placement placement, const std::string& line)
+	constexpr auto changeable_at_runtime = false;
+	const auto prop = conf->AddSection_prop("drive", nullptr, changeable_at_runtime);
+
+	// Define the allowed keys and types
+	constexpr auto on_startup = Property::Changeable::OnlyAtStart;
+	prop->Add_string("type", on_startup, "")
+	        ->Set_values({"dir", "floppy", "cdrom", "overlay"});
+	prop->Add_string("label", on_startup, "");
+	prop->Add_string("path", on_startup, "");
+	prop->Add_string("override_drive", on_startup, "");
+	prop->Add_bool("verbose", on_startup, true);
+	prop->Add_bool("readonly", on_startup, false);
+
+	return conf;
+}
+
+std::optional<AutoExecModule::AutoMountSettings> AutoExecModule::ParseDriveConf(
+        const std_fs::path& conf_path)
+{
+	AutoMountSettings settings = {};
+
+	// If the conf path doesn't exist, at least return the default quiet arg
+	if (!path_exists(conf_path)) {
+		return settings;
+	}
+
+	// If we couldn't parse it, return the defaults
+	const auto conf = SpecifyDriveConf();
+	assert(conf);
+	if (!conf->ParseConfigFile("auto-mounted drive", conf_path.string())) {
+		return settings;
+	}
+
+	const auto section = static_cast<Section_prop*>(conf->GetSection("drive"));
+
+	const auto override_drive = section->Get_string("override_drive");
+	if (override_drive.length() == 1 && override_drive[0] >= 'a' &&
+	    override_drive[0] <= 'y') {
+		settings.override_drive = override_drive;
+	} else if (!override_drive.empty()) {
+		LOG_ERR("AUTOMOUNT: %s: setting 'override_drive = %s' is invalid",
+		        conf_path.c_str(),
+		        override_drive.c_str());
+		LOG_ERR("AUTOMOUNT: The override_drive setting can be left empty or a drive letter from 'a' to 'y'");
+	}
+
+	settings.readonly = section->Get_bool("readonly");
+	settings.label    = section->Get_string("label");
+	settings.type     = section->Get_string("type");
+	settings.path     = section->Get_string("path");
+	settings.verbose  = section->Get_bool("verbose");
+
+	return settings;
+}
+
+std::string AutoExecModule::BuildAutoMountFolderCommand(
+        const std::string_view dir_letter, const std_fs::path& drive_path,
+        const std::optional<AutoMountSettings>& settings)
+{
+	auto command = CmdMount;
+
+	if (!settings.has_value() || settings->override_drive.empty()) {
+		command += dir_letter;
+	} else {
+		command += settings->override_drive;
+	}
+
+	command += " " + Quote + simplify_path(drive_path).string() + Quote;
+
+	if (settings.has_value()) {
+		if (!settings->type.empty()) {
+			command += " -t " + settings->type;
+		}
+
+		if (!settings->label.empty()) {
+			command += " -label " + settings->label;
+		}
+
+		if (settings->readonly) {
+			command += " -ro";
+		}
+
+		if (!settings->verbose) {
+			command += ToNul;
+		}
+	}
+
+	return command;
+}
+
+std::vector<std_fs::path> AutoExecModule::GetMountableImagesPaths(const std_fs::path& drive_path)
+{
+	std::vector<std_fs::path> image_paths;
+
+	for (const auto& entry : std_fs::directory_iterator(drive_path)) {
```

Comment:
The directory content order is unspecified, right?
Wouldn't it be better to sort it, so that `FooBarGame_CD1.iso` is going to be mounted as the 1st image, `FooBarGame_CD2.iso` as the 2nd, etc.?

Reply:
Yeah, I'm pretty sure you'll need to sort it (check out how the FluidSynth SF2 resolution works, that must be A-Z sorted).